1 - DataTypes/Code/WeirdCalculator.cs

```diff
@@ -39,29 +41,47 @@ public static decimal ComplexDecimalAdd(float firstValue, double secondValue, de
 			 * I will give you one clue / tip - You *will* need to handle exceptions for the final solution.
 			 * If you are new to exceptions, google "C# Exception Handling" - litterally half our jobs is googling :P
 			 */
-
-			throw new NotImplementedException();
+            decimal firstValueDecimal;
+            decimal secondValueDecimal;
```

Comment:
When declaring variables like this, it is always wise to declare a default value for them, (usually zero). Makes the code more explicit and reduces errors down the line while the code changes.